controllers/adapters_controller.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * Adapter Controller.
+ *
+ * Manages HTTP requests to /adapters. 
+ *  
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+var express = require('express');
+var adapterManager = require('../adapter-manager');
+
+var adaptersController = express.Router();
+
+/**
+ * Return a list of adapters
+ */
+adaptersController.get('/', (request, response) => {
+  adapterManager.discoverAdapters().then(function(adapters) {
+    var adapterList = [];
+    for (var adapter of adapters) {
+      adapterList.push(adapter.asDict());
+    }
+    response.json(adapterList);
+  });
+});
+
+/**
+ * Cancel Action: add a new device to the adapter.
+ */
+adaptersController.delete('/addSomeDevice', (request, response) => {
+  adapterManager.cancelAddSomeDevice();
+  response.status(204).send();
+});
+
+/**
+ * Cancel Action: remove a device from the adapter.
+ */
+adaptersController.delete('/removeSomeDevice', (request, response) => {
+  adapterManager.cancelRemoveSomeDevice();
+  response.status(204).send();
+});
+
+/**
+ * Get a particular adapter.
+ */
+adaptersController.get('/:adapterId/', (request, response) => {
+  var adapterId = request.params['adapterId'];
+
+  if (adapterId === 'addSomeDevice') {
+    // Action: add a new device to the adapter.
+    adapterManager.addSomeDevice().then((device) => {
+      console.log('addSomeDevice added id', device.getId(), 'name', device.getName());
+    }, (str) => {
+      console.log('addSomeDevice cancelled');
+    });
+    response.json({
+      'name': 'addSomeDevice',
+      'href': request.originalUrl,
+    });
+    return
+  }
+
+  if (adapterId === 'removeSomeDevice') {
+    // Action: remove a device from the adapter.
+    adapterManager.removeSomeDevice().then((device) => {
+      console.log('removeSomeDevice removed id', device.getId(), 'name', device.getName());
+    }, (str) => {
+      console.log('removeSomeDevice cancelled');
+    });
+    response.json({
+      'name': 'removeSomeDevice',
+      'href': request.originalUrl,
+    });
+    return;
+  }
+
+  var adapter = adapterManager.getAdapter(adapterId);
+  if (adapter) {
+    response.json(adapter.asDict());
+    return;
+  }
+  response.status(404).send('Adapter "' + adapterId + '" not found.');
+});
+
+/**
+ * Get a list of devices associated with a particular adapter.
+ */
+adaptersController.get('/:adapterId/devices/', (request, response) => {
```

Comment:
We could make it so that the adaptersController is renamed to debug/test and only shows up if a specific argument is passed when we start the app. That reinforces the fact that it's really just for test/debug.